components/PullToRefresh.vue

```diff
@@ -0,0 +1,147 @@
+<docs>
+	# 下拉刷新
+
+	FIXME: 当前因为使用了TouchEvent只支持Chromium内核的浏览器，请考虑使用PointerEvent但是注意浏览器默认触摸事件的冲突问题。
+</docs>
+
+<script setup lang="ts">
+	const props = withDefaults(defineProps<{
+		/** 滚动容器 querySelector。 */
+		scrollContainerSelector?: string;
+		/** 滚动内容 querySelector。 */
+		scrollContentSelector?: string;
+	}>(), {
+		scrollContainerSelector: "#mainScroller",
+		scrollContentSelector: "#mainScroller > .content",
+	});
+	const refreshing = defineModel<boolean>();
+
+	const scrollContainer = ref<HTMLElement | null>(null);
+	const scrollContent = ref<HTMLElement | null>(null);
```

Comment:
```suggestion
	const scrollContainer = ref<HTMLElement>();
	const scrollContent = ref<HTMLElement>();
```
据我所知 null 是多余的，官网的教程多做的这一步是纯扯淡的。

Reply:
不这么写类型会报错，你可以试试。

Reply:
> 不这么写类型会报错，你可以试试。

了解。